lib/dry/system/booter.rb

```diff
@@ -69,6 +69,17 @@ def finalize!
         freeze
       end
 
+      # @api private
+      def shutdown
+        components.each do |component|
+          begin
+            stop(component)
+          rescue ComponentNotStartedError
+            next
+          end
+        end
+      end
```

Comment:
IIRC there's a way to check which components have been started. Maybe you could use that instead of rescuing from `ComponentNotStartedError`?

Reply:
@solnic Yeah makes much more sense. Looks like I have fixated mind about the rescue block. Thank you for the patients.

Done https://github.com/dry-rb/dry-system/pull/98/commits/bec81caf8d1d25051f659ab461ee8a72f894ae99